File: base/src/main/java/com/lege/android/base/util/JSON.java
package com.lege.android.base.util;

import com.blankj.utilcode.util.GsonUtils;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

/**
 * Description:
 * Created by loctek on 2021/11/1.
 */
public class JSON {
    public static <T> T parseObject(String json,Class<T> clz){
        return GsonUtils.fromJson(json,clz);
    }
    public static <T> List<T> parseArray(String json){
        Type type = new TypeToken<List<T>>(){}.getType();
        return GsonUtils.fromJson(json,type);
    }
    public static String toJSONString(Object object){
        return GsonUtils.toJson(object);
    }

}

File: base/src/main/java/com/lege/android/base/util/JSON.kt
package com.lege.android.base.util

import com.blankj.utilcode.util.GsonUtils
import com.google.gson.reflect.TypeToken

/**
 * Description:
 * Created by loctek on 2021/11/1.
 */
inline fun <reified T> String.parseObject(): T = GsonUtils.fromJson(this, T::class.java)
inline fun <reified E> E.toJsonString(): String = GsonUtils.toJson(this,E::class.java)
inline fun <reified E> String.parseArray(): List<E> =
    GsonUtils.fromJson<List<E>>(this, object : TypeToken<List<E>>() {}.type)
